### ISSUE TITLE
OpenMP wrong-code issue with pointers in an OMP loop

### ISSUE THREAD

**Issue by janusw** [flang-compiler/flang#345]
Consider the following test case:

```
program omp_bug
  integer, parameter :: cnt = 100
  integer, pointer :: bnd(:,:)
  integer, pointer :: chamber(:)
  integer :: i, c

  allocate(bnd(1:2,cnt))
  do i = 1, cnt
    bnd(1:2, i) = (/ i, i+1 /)
  end do

  allocate(chamber(1:cnt+1), source=0)

!$omp parallel do private(i)
  do i = 1,cnt
    chamber(bnd(1:2,i)) = 1
  end do
!$omp end parallel do

  do i = 1, cnt+1
    c = chamber(i)
    if (c <= 0) then
      print *, i, c
      call abort()
    end if
  end do

end
```

Compiling this with `flang -mp` and executing with `OMP_NUM_THREADS>1` yields errors like:

```
           39            0
Aborted (core dumped)
```

The first number varies, the second is always zero, so apparently something goes wrong in the loop that sets the chamber array.

It runs through error-free with  `OMP_NUM_THREADS=1` or when making one of the POINTERs an ALLOCATABLE. Moreover it runs fine with gfortran and ifort (also multi-threaded).

**Comment by janusw** [comment]
Note that flang seems to share this bug with PGI 17.10.

**Comment by janusw** [comment]
I also reported this in the PGI forum, see here:
http://www.pgroup.com/userforum/viewtopic.php?t=5887

**Comment by janusw** [comment]
Seems like this error is gone on current master (e8123d1aa7882ea996851133b77ef17f1ec0e51d), if I'm not mistaken ...

**Comment by janusw** [comment]
Apparently it has been fixed by f06ba56e33cdf5979ec4a425cf6acea098e8635b. Thanks! :D
